pincer/objects/guild/permissions.py

```diff
@@ -0,0 +1,272 @@
+    add_reactions: Optional[bool] = None
+    view_audit_log: Optional[bool] = None
+    priority_speaker: Optional[bool] = None
+    stream: Optional[bool] = None
+    view_channel: Optional[bool] = None
+    send_messages: Optional[bool] = None
+    send_tts_messages: Optional[bool] = None
+    manage_messages: Optional[bool] = None
+    embed_links: Optional[bool] = None
+    attach_files: Optional[bool] = None
+    read_message_history: Optional[bool] = None
+    mention_everyone: Optional[bool] = None
+    use_external_emojis: Optional[bool] = None
+    view_guild_insights: Optional[bool] = None
+    connect: Optional[bool] = None
+    speak: Optional[bool] = None
+    mute_members: Optional[bool] = None
+    deafen_members: Optional[bool] = None
+    move_members: Optional[bool] = None
+    use_vad: Optional[bool] = None
+    change_nickname: Optional[bool] = None
+    manage_nicknames: Optional[bool] = None
+    manage_roles: Optional[bool] = None
+    manage_webhooks: Optional[bool] = None
+    manage_emojis_and_stickers: Optional[bool] = None
+    use_application_commands: Optional[bool] = None
+    request_to_speak: Optional[bool] = None
+    manage_events: Optional[bool] = None
+    manage_threads: Optional[bool] = None
+    create_public_threads: Optional[bool] = None
+    create_private_threads: Optional[bool] = None
+    use_external_stickers: Optional[bool] = None
+    send_messages_in_threads: Optional[bool] = None
+    start_embedded_activities: Optional[bool] = None
+    moderate_members: Optional[bool] = None
+
+    def __setattr__(self, name: str, value: Optional[bool]) -> None:
+        if not isinstance(value, bool) and value is not None:
+            raise ValueError(f"Permission {name!r} must be a boolean or None")
+        return super().__setattr__(name, value)
+
+    def __eq__(self, object) -> bool:
+        """
+        Permission equality is determined by comparing the integer values of the permissions
+        """
+        if isinstance(object, Permission):
+            return self.to_int() == object.to_int()
+        elif isinstance(object, tuple):
+            return self.to_int() == object
+
+        return False
+
+    @classmethod
+    def from_int(cls, allow: int, deny: int) -> Permission:
+        """
+        Create a Permission object from an integer representation of the permissions (deny and allow)
+
+        Parameters
+        __________
+        allow: :class:`int`
+            The integer representation of the permissions that are allowed
+        deny: :class:`int`
+            The integer representation of the permissions that are denied
+        """
+        clsobj = cls()
+
+        for enum in Permissions:
+            if bool(enum.value & allow):
+                setattr(clsobj, enum.name.lower(), True)
+            elif bool(enum.value & deny):
+                setattr(clsobj, enum.name.lower(), False)
+            else:
+                setattr(clsobj, enum.name.lower(), None)
```

Comment:
```suggestion
        for enum in Permissions:
            if enum.value & allow:
                setattr(clsobj, enum.name.lower(), True)
            elif enum.value & deny:
                setattr(clsobj, enum.name.lower(), False)
            else:
                setattr(clsobj, enum.name.lower(), None)
```
Or even this (the parentheses at L225 ain't needed but make it slightly more readable)
```suggestion
        for enum in Permissions:
            setattr(
                clsobj, enum.name.lower(),
                True if enum.value & allow else
                (False if enum.value & deny else None)
            )
```

Reply:
Maybe not the shortest but way more readable

```suggestion
        for enum in Permissions:
            value = None

            if enum.value & allow:
                value = True

            elif enum.value & deny:
                value = False

            setattr(clsobj, enum.name.lower(), value)
```